src/docfx/lib/HostNameUtility.cs

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Docs.Build;
+internal class HostNameUtility
+{
+    public static string MigrateHostName(string hostName, Dictionary<string, string>? hostNameMapping)
+    {
+        if (hostNameMapping is null || string.IsNullOrEmpty(hostName))
+        {
+            return hostName;
+        }
+
+        if (hostNameMapping.ContainsKey(hostName))
+        {
+            return hostNameMapping[hostName];
+        }
+        return hostName;
+    }
+
+    public static string? MigrateHostForUrl(string? url, Dictionary<string, string>? hostNameMapping)
+    {
+        if (hostNameMapping is null || string.IsNullOrEmpty(url))
+        {
+            return url;
+        }
+
+        if (Uri.IsWellFormedUriString(url, UriKind.Absolute))
+        {
+            var uri = new Uri(url);
+            if (hostNameMapping.ContainsKey(uri.Host))
```

Comment:
This logic is duplicate with the one in `MigrateHostName` -> suggest to drop the `if` here